engine/src/main/coffee/extensions/table.coffee

```diff
@@ -0,0 +1,218 @@
+        "(a table key may only be a number, a string, true or false, or a list whose items are valid keys)")
+
+
+module.exports = {
+
+  dumper: { canDump: isTable, dump: (x) -> "{{table: #{dumpTable(x)}}}" }
+
+  init: (workspace) ->
+
+    # () => Table
+    make = ->
+      new Map()
+
+    # (List) => Map
+    fromList = (list) ->
+      checkIsValidList(list)
+      new Map(list.slice(0))
+
+    # (Table) => List
+    toList = (table) ->
+      checkInput({table: table})
+      Array.from(table).slice(0)
+
+    # (Table) => Unit
+    clear = (table) ->
+      checkInput({table: table})
+      table.clear()
+      return
+
+    # (Table, Any) => Any
+    get = (table, key) ->
+      checkInput({table: table})
+      if key not instanceof Array
+        return table.get(key)
+
+      originKey = getOriginKey(table, key)
+      if originKey?
+        table.get(originKey)
+      else
+        throw new Error("Extension Exception: No value for #{key} in table.")
+
+    # (Table, Any) => Boolean
+    hasKey = (table, key) ->
+      checkInput({table: table})
+      if key not instanceof Array
+        return table.has(key)
+
+      Array.from(table.keys()).some((k) -> equals(k, key))
+
+    # (Table) => List
+    keys = (table) ->
+      checkInput({table: table})
+      Array.from(table.keys()).slice(0)
+
+    # (Table) => List
+    values = (table) ->
+      checkInput({table: table})
+      Array.from(table.values()).slice(0)
+
+    # (Table) => Number
+    length = (table) ->
+      checkInput({table: table})
+      table.size
+
+    # (Table, Any, Any) => Unit
+    put = (table, key, value) ->
+      checkInput({table: table, key: key})
+
+      if key not instanceof Array
+        table.set(key, value)
+        return
+
+      key = getOriginKey(table, key) ? key
+      table.set(key, value)
+      return
+
+    # (Table, Any) => Unit
+    remove = (table, key) ->
+      checkInput({table: table})
+
+      if key not instanceof Array
+        table.delete(key)
+        return
+
+      originKey = getOriginKey(table, key)
+      if originKey?
+        table.delete(originKey)
+
+      return
+
+    # (List) => Table
+    counts = (list) ->
+      counts = new Map()
+
+      for item in list
+        count = if hasKey(counts, item) then get(counts, item) + 1 else 1
+        put(counts, item, count)
+
+      counts
+
+    # (Table, Any, Any) => Any
+    getOrDefault = (table, key, defaultValue) ->
+      checkInput({table: table})
+
+      if hasKey(table, key)
+        get(table, key)
+      else
+        defaultValue
+
+    # (Agentset, Reporter) => Table
+    groupAgents = (agentset, reporter) ->
+      group = new Map()
+
+      agentset.shufflerator().forEach( (agent) ->
+        key = workspace.world.selfManager.askAgent(reporter)(agent)
+        value = group.get(key) ? []
+        value.push(agent)
+        group.set(key, value)
+      )
+
+      group.forEach( (value, key, map) ->
```

Comment:
Since `map` isn't used here you can just omit it, `(value, key) -> ...`

Reply:
Okay, I've fixed it.